cdap-watchdog/src/main/java/co/cask/cdap/logging/save/KafkaMessageCallback.java

```diff
@@ -100,11 +92,19 @@ public void onReceived(Iterator<FetchedMessage> messages) {
                   message.getTopicPartition().getTopic(),
                   message.getTopicPartition().getPartition());
       }
-
-      count++;
     }
 
-    if (count > 0) {
+    int count = events.size();
+    if (!events.isEmpty()) {
+      for (KafkaLogProcessor processor : kafkaLogProcessors) {
+        try {
+          processor.process(events.iterator());
```

Comment:
Each processor is independent and should process all the events from the iterator. For example, one processor writes events to files and another processor can emit metrics.